test/integration/select_test.lua

```diff
@@ -77,6 +81,111 @@ pgroup.test_not_valid_value_type = function(g)
     t.assert_str_contains(err.err, "Supplied key type of part 0 does not match index part type: expected unsigned")
 end
 
+local select_safety_cases = {
+    nil_and_nil_opts = {
+        has_crit = true,
+        user_conditions = nil,
+        opts = nil,
+    },
+    fullscan_false = {
+        has_crit = true,
+        user_conditions = nil,
+        opts = {fullscan = false},
+    },
+    fullscan_true = {
+        has_crit = false,
+        user_conditions = nil,
+        opts = {fullscan = true},
+    },
+    limit = {
+        has_crit = false,
+        user_conditions = nil,
+        opts = {first = 123},
+    },
+    min_limit = {
```

Comment:
Well, it's more like max negative limit

Reply:
No problems. I've renamed it to `max_negative_limit`.